### PR TITLE
chore(ci): use new contexts - DIA-59103

### PR DESCRIPTION
**Pull request by arththebird**
## Description 

Remove gemfury context in favor of new contexts.

Update orbs to latest.

## Related JIRA issues

* [DIA-59103]

## Validation

Ci still works ✅ 

[DIA-59103]: https://dialoguemd.atlassian.net/browse/DIA-59103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -100,7 +104,7 @@ jobs:
             - run:
                 name: "install tox"
                 command: |
-                  poetry run pip install -U tox
+                  poetry run pip install -U tox==4.7.0
```
Comment (arththebird) [inline .circleci/config.yml:107]: latest version breaks the pipeline

File: .circleci/config.yml [hunk 2]
```diff
@@ -51,15 +56,14 @@ workflows:
     jobs:
       - publish:
           name: publish
-          context: gemfury-download # Needed to build, the upload token is in the CI env vars
+          context: code-validator # Needed to build, the upload token is in the CI env vars
```
Comment (edouardlp) [inline .circleci/config.yml:59]: Is the comment still true?
```suggestion
          context: code-validator
```

**Review by edouardlp — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (arththebird) [reply .circleci/config.yml:59]: yes, because it's public, we upload to pypi and the token is in the CI env vars :)

Comment (arththebird) [reply .circleci/config.yml:59]: I guess I can just remove it though 🤷

**Pull request merged**